ufl/core/external_operator.py

```diff
@@ -0,0 +1,278 @@
+            sub_element = sub_element.sub_elements()[0]
+
+        if len(s) == 0:
+            ufl_element = sub_element
+        elif len(s) == 1:
+            ufl_element = VectorElement(sub_element)
+        else:
+            ufl_element = TensorElement(sub_element, shape=s)
+        domain = default_domain(ufl_element.cell())
+        return FunctionSpace(domain, ufl_element)
+
+    def evaluate(self, x, mapping, component, index_values):
+        """Evaluate expression at given coordinate with given values for terminals."""
+        error("Symbolic evaluation of %s not available." % self._ufl_class_.__name__)
+
+    def _ufl_expr_reconstruct_(self, *operands, function_space=None, derivatives=None, count=None, coefficient=None, arguments=None):
+        "Return a new object of the same type with new operands."
+        deriv_multiindex = derivatives or self.derivatives
+
+        if deriv_multiindex != self.derivatives:
+            # If we are constructing a derivative
+            corresponding_count = None
+            corresponding_coefficient = None
+            e_master = self._extop_master
+            for ext in e_master._extop_dependencies:
+                if ext.derivatives == deriv_multiindex:
+                    return ext._ufl_expr_reconstruct_(*operands, function_space=function_space,
+                                                      derivatives=deriv_multiindex, count=count,
+                                                      coefficient=coefficient,
+                                                      arguments=arguments)
+        else:
+            corresponding_count = self.count()
+            corresponding_coefficient = coefficient or self._coefficient
+
+        reconstruct_op = type(self)(*operands, function_space=function_space or self.ufl_function_space(),
+                                    derivatives=deriv_multiindex,
+                                    count=corresponding_count,
+                                    coefficient=corresponding_coefficient,
+                                    arguments=arguments or (self.arguments() + self.action_args()))
+
+        if deriv_multiindex != self.derivatives:
+            # If we are constructing a derivative
+            self._extop_master._extop_dependencies.append(reconstruct_op)
+            reconstruct_op._extop_master = self._extop_master
+        else:
+            reconstruct_op._extop_master = self._extop_master
+            reconstruct_op._extop_dependencies = [e._ufl_expr_reconstruct_(*operands) for e in self._extop_dependencies[1:]]
+        return reconstruct_op
+
+    def _add_dependencies(self, derivatives, args):
+        v = list(self._ufl_expr_reconstruct_(*self.ufl_operands, derivatives=d, arguments=a)
+                 for d, a in zip(derivatives, args))
+        self._extop_master._extop_dependencies.extend(x for x in v if x not in self._extop_master._extop_dependencies)
+        return self
```

Comment:
This dependency stuff smells very "I want to compute something". Does it belong in the symbolics?

Reply:
See comment above